File: Set-1/Palindrome.java
import java.util.Scanner;

class Palindrome {
    public static void main(String args[]) {
        Scanner scan = new Scanner(System.in);
        System.out.println("Enter a number");
        int num = scan.nextInt();
        System.out.println("The number " + num + " is " + (isPalindrome(num) ? "palindrome" : "not palindrome"));
    }

    static boolean isPalindrome(int n) {
        int rev = 0, temp = n, rem;
        do {
            rem = temp % 10;
            rev = rev * 10 + rem;
            temp /= 10;
        } while (temp != 0);
        if (rev == n)
            return true;
        else
            return false;
    }
}

File: Class/final_exam/palindrome.c
#include <stdio.h>
int main()
{
    int num;
    printf("Enter a number\t");
    scanf("%d", &num);
    int n = num, rev = 0;
    while (n != 0)
    {
        int rem = n % 10;
        rev = rev * 10 + rem;
        n /= 10;
    }
    printf("The number %d is %s\n", num, num == rev ? "palindrome" : "not palindrome");
    return 0;
}